stub/src/main/java/io/grpc/stub/ServerCalls.java

```diff
@@ -76,9 +76,6 @@ public void onPayload(ReqT request) {
               // We delay calling method.invoke() until onHalfClose(), because application may call
               // close(OK) inside invoke(), while close(OK) is not allowed before onHalfClose().
               this.request = request;
-
-              // Request delivery of the next inbound message.
-              call.request(1);
```

Comment:
Hmm... So before if the type was wrong we would end up failing the call. Now we will deadlock. I don't know how I feel about that yet...